vnext/Microsoft.ReactNative/ReactApplication.cpp

```diff
@@ -96,7 +98,12 @@ void ReactApplication::JavaScriptBundleFile(hstring const &value) noexcept {
 void ReactApplication::OnActivated(Windows::ApplicationModel::Activation::IActivatedEventArgs const &e) {
   if (e.Kind() == Windows::ApplicationModel::Activation::ActivationKind::Protocol) {
     auto protocolActivatedEventArgs{e.as<Windows::ApplicationModel::Activation::ProtocolActivatedEventArgs>()};
-    ::Microsoft::ReactNative::LinkingManagerModule::OpenUri(protocolActivatedEventArgs.Uri());
+    const auto uri = protocolActivatedEventArgs.Uri();
+    const auto properties = InstanceSettings().Properties();
+    if (LinkingHelper::InitialUrl(properties).empty()) {
```

Comment:
wouldn't you always want to set the initial url here? I'm curious what usage the initial url has here, if you are navigating to it via OpenUri anyway

Reply:
I was keeping the behavior the same as it was before. Previously, there was a static variable inside `LinkingManagerModule` that checked if the initial URI was not set yet, then set the first value passed to OpenUri as the initial URI (then never again). This was obviously brittle because it was a static value, and if you have multiple instances, you'd presumably want a per instance initial URL.

This emulates the same behavior but doesn't necessarily require it. I looked at RN Android functionality, and it appears that it just gets the current Activity and pulls the URI from it's Intent data. I'm not an expert in app lifecycle for Android, but I'm guessing that the current Activity is whatever was invoked most recently, so it would follow that your suggestion is sound - we should just always set the InitialUrl any time the app is protocol activated.

I'm guessing the same is true for iOS. I think the name of the method `Linking.getInitialUrl` is misleading. Perhaps it should be `Linking.getLatestUrl` or something.

Reply:
And actually, we'll need to fix the behavior for multiple instances as well to support app-specific routing of `url` events to specific instances. Currently `OpenUrl` triggers on all instances, but we'll likely want to be able to invoke it on a specific instance (or subset of instances).

I'm weary of making a change to either the "set once" behavior for initial URL or the pattern of emitting the `url` event for all instances because there may be apps that depend on it. Though presumably we could just document the breaking change and ask folks to refactor there deep linking a bit...

Reply:
Actually, this is already a breaking change since it will require folks depending on `getInitialUrl` to actually set the property on the ReactPropertyBag. I'll re-work this a bit to make it non-breaking, keep the static initial URL variable, and prefer the property bag value if it's been set.